diff --git a/node_modules/gatsby/dist/commands/serve.js b/node_modules/gatsby/dist/commands/serve.js
index 9e5ec6a..3466efd 100644
--- a/node_modules/gatsby/dist/commands/serve.js
+++ b/node_modules/gatsby/dist/commands/serve.js
@@ -93,7 +93,7 @@ module.exports = async program => {
   const router = express.Router();
   app.use(telemetry.expressMiddleware(`SERVE`));
   router.use(compression());
-  router.use(express.static(`public`));
+  router.use(express.static(`public`, { extensions: [`html`] }));
   const matchPaths = await readMatchPaths(program);
   router.use(matchPathRouter(matchPaths, {
     root
diff --git a/node_modules/gatsby/dist/utils/worker/render-html.js b/node_modules/gatsby/dist/utils/worker/render-html.js
index cda957f..a1d101e 100644
--- a/node_modules/gatsby/dist/utils/worker/render-html.js
+++ b/node_modules/gatsby/dist/utils/worker/render-html.js
@@ -14,7 +14,7 @@ const generatePathToOutput = outputPath => {
   let outputFileName = outputPath.replace(/^(\/|\\)/, ``); // Remove leading slashes for webpack-dev-server
 
   if (!/\.(html?)$/i.test(outputFileName)) {
-    outputFileName = path.join(outputFileName, `index.html`);
+    outputFileName = outputFileName + `${outputFileName.endsWith('/') ? 'index' : ''}.html`;
   }
 
   return path.join(process.cwd(), `public`, outputFileName);
